### ISSUE TITLE
Let pipelines use model from PVC

### ISSUE THREAD

**Issue by piotrpdev** [opendatahub-io/ai-edge#87]
## Description

You could provide a local model back in #18 (https://github.com/piotrpdev/azureml-model-to-edge/commit/626d5a5c85ff8c5cdd4019714dce91fe1c1b7751) but that was replaced for the S3 method (and also Git in #57 soon). Ideally a single task should handle getting the model and take some input for how e.g. using Git, S3, Azure, PVC (local), etc. Modifying our existing [`kserve-download-model`](https://github.com/opendatahub-io/ai-edge/blob/e0086c9e2c223c8d4a1df7a415d8aee736365e75/pipelines/tekton/azureml-container-pipeline/kserve-download-model.yaml) Tekton Task is probably best for this.

Since no downloading needs to actually be done for the PVC (local) scenario, you could maybe just skip the `kserve-download-model` Task all together using a [Tekton `when` guard](https://tekton.dev/docs/pipelines/pipelines/#guard-task-execution-using-when-expressions). However, I think modifying the Task to move the model from the provided PVC to the [`buildah-cache`](https://github.com/opendatahub-io/ai-edge/blob/e0086c9e2c223c8d4a1df7a415d8aee736365e75/pipelines/tekton/azureml-container-pipeline/azureml-container-pipeline.yaml#L72) workspace is better, especially since we're probably going to make each `PipelineRun` use [its own PVC soon](https://github.com/opendatahub-io/ai-edge/pull/54#discussion_r1319746380) (using [`volumeClaimTemplate`](https://tekton.dev/docs/pipelines/workspaces/#volumeclaimtemplate)).

<br>

> _Isn't downloading from Git enough?_

The PoC phase two requirements specify using a model stored in a PVC as a _'P0: must have'_ [[1]](https://docs.google.com/document/d/1C47Jw4EC8KNvi01ZKG0Rd8spPBJ0nIx-FbYal20EApc/edit#heading=h.e7hnzf726tke). When RHODS integration and the new Model Registry [[2]](https://docs.google.com/document/d/1G-pjdGaS2kLELsB5kYk_D4AmH-fTfnCnJOhJ8xENjx0/edit?usp=sharing) comes into play this will probably be an important feature.

## A/C

* Provide template PVC file which uses best practices
* Easy and documented way of uploading a local model to the PVC e.g. using a ubi9 pod and `oc cp`
* Have a step/task that can copy the model from one PV (e.g. one being used by a Jupyter Notebook) to the PV/workspace being used by the pipelines.
* `azureml-container-pipeline` still works when using:
  * PVC (local model)
  * S3
  * Git

**Comment by adelton** [comment]
> You could provide a local model back in #18 ([piotrpdev/azureml-model-to-edge@626d5a5](https://github.com/piotrpdev/azureml-model-to-edge/commit/626d5a5c85ff8c5cdd4019714dce91fe1c1b7751)) but that was replaced for the S3 method (and also Git in #57 soon).

I wonder -- if this is a PoC, shouldn't it showcase all methods? Have three (or how many needed) models and applications, one with model in the git, one in S3 pulled in in build-time, and one in S3 pulled in in run-time? Unless we have a strong reason to avoid any of those approaches, we shouldn't we replacing them, we should be adding the new options.

And in documentation explain how they differ.

> Ideally a single task should handle getting the model and take some input for how e.g. using Git, S3, Azure, PVC (local), etc.

Overall though, this is not Edge-specific, is it? You might want something like that in the general OpenDataHub case as well.

**Comment by piotrpdev** [comment]
> we shouldn't we replacing them, we should be adding the new options.
I agree, I think I removed the local model method by mistake or assumed we weren't going to use it at the time.

> You might want something like that in the general OpenDataHub case as well.
Can you elaborate on this please?

**Comment by adelton** [comment]
The various approaches to connecting models to runtimes might go to whatever demos / guides / learning materials that we have about using Open Data Hub and building pipelines there. We we have a place to carry the information outside of this `ai-edge` repo, we wouldn't need to focus on the wealth of options here.

**Comment by piotrpdev** [comment]
I'm sorry if I didn't make it clear enough, but for the issue to be resolved the pipelines need to be able to use a model from _any_ PVC. Ideally this means we:

1. Provide a PVC template along with a way to upload a local model to it (like #115)
1. Have a step/task that can copy the model from one PV (e.g. one being used by a Jupyter Notebook) to the PV/workspace being used by the pipelines.
